pkg/api/server/v1alpha2/records_test.go

```diff
@@ -557,3 +557,48 @@ func TestUpdateRecord(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteRecord(t *testing.T) {
+	srv, err := New(test.NewDB(t))
+	if err != nil {
+		t.Fatalf("failed to create temp file for db: %v", err)
+	}
+
+	ctx := context.Background()
+	result, err := srv.CreateResult(ctx, &pb.CreateResultRequest{
+		Parent: "foo",
+		Result: &pb.Result{
+			Name: "foo/results/bar",
+		},
+	})
+	if err != nil {
+		t.Fatalf("CreateResult: %v", err)
+	}
+	r, err := srv.CreateRecord(ctx, &pb.CreateRecordRequest{
+		Parent: result.GetName(),
+		Record: &pb.Record{
+			Name: recordutil.FormatName(result.GetName(), "baz"),
+		},
+	})
+	if err != nil {
+		t.Fatalf("CreateRecord(): %v", err)
+	}
+
+	t.Run("success", func(t *testing.T) {
+		// Delete inserted record
+		if _, err := srv.DeleteRecord(ctx, &pb.DeleteRecordRequest{Name: r.GetName()}); err != nil {
+			t.Fatalf("could not delete record: %v", err)
+		}
+		// Check if the the record is deleted
+		if r, err := srv.GetRecord(ctx, &pb.GetRecordRequest{Name: r.GetName()}); err == nil {
+			t.Fatalf("expected record to be deleted, got: %+v", r)
+		}
```

Comment:
This would pass if GetRecord returned any error, including other errors besides NOT_FOUND. We probably want to do something like this instead:

```suggestion
		if r, err := srv.GetRecord(ctx, &pb.GetRecordRequest{Name: r.GetName()}); status.Code(err) != codes.NotFound {
			t.Fatalf("expected record to be deleted, got: %+v, %v", r, err)
		}
```